pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -45,11 +46,15 @@ var _ = Describe("MachineSet Reconciler", func() {
 		err = add(mgr, reconciler, reconciler.MachineToMachineSets)
 		Expect(err).NotTo(HaveOccurred())
 
+		var mgrCtx context.Context
+		mgrCtx, mgrCtxCancel = context.WithCancel(ctx)
+		mgrStopped = make(chan struct{})
```

Comment:
I'm not following how this is going to work.  We create mgrStopped channel, and subsequently close it, but it doesn't seem to be used by the manager in any way?

Reply:
This may not solve the whole of the bug if that is the case, but it takes a step towards it by ensuring that the test only completes once the stop channel is closed.

Previously we could have the following:
- Close context
- Manager starts to shutdown
- Test case ends
- New test starts
- Manager finishes shutdown

With this it should now be:
- Close context
- Manager starts to shutdown
- Manager finishes shutdown
- Test case ends
- New test starts

It was my understanding that this was at least part of the problem being highlighted in the bug?

Reply:
Okay, I'm grokking what mgrStopped is doing now.  We ensure that we wait for the manager to actually return, whereas we weren't before (we would just complain if it failed to stop clean later).

This might improve what we're seeing, depending on how the bug actually manifests.  I can't recall if it was complaining about reconciling another test's objects or not, but I know that is potentially one cause.